File: src/main/java/io/mfedirko/learning/UpdateLessonForm.java
package io.mfedirko.learning;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import lombok.extern.jackson.Jacksonized;

import java.time.LocalDateTime;

@Getter
@Builder
@Value
@Jacksonized
public class UpdateLessonForm {
    LocalDateTime creationTimestamp;

    @With
    Long creationTimestampMillis;

    @NotEmpty
    String category;

    @NotEmpty
    String title;

    @NotEmpty
    String description;

    public static UpdateLessonForm fromLesson(Lesson lesson) {
        return UpdateLessonForm.builder()
                .creationTimestampMillis(lesson.getCreationTimestampMillis())
                .creationTimestamp(lesson.getCreationTimestamp())
                .title(lesson.getTitle())
                .category(lesson.getCategory())
                .description(lesson.getDescription())
                .build();
    }
}

File: src/main/kotlin/io/mfedirko/learning/UpdateLessonForm.kt
package io.mfedirko.learning

import jakarta.validation.constraints.NotEmpty

class UpdateLessonForm {
    @NotEmpty
    var category: String? = null

    @NotEmpty
    var title: String? = null

    @NotEmpty
    var description: String? = null

    companion object {
        @JvmStatic
        fun fromLesson(lesson: Lesson): UpdateLessonForm {
            return UpdateLessonForm().apply {
                title = lesson.title
                category = lesson.category
                description = lesson.description
            }
        }
    }
}